airflow-core/src/airflow/ui/src/i18n/check_translations_completeness.py

```diff
@@ -235,6 +245,123 @@ def get_outdated_entries_for_language(
+            file_path = Path(LOCALES_DIR / lang / filename)
+            # Always get total from English version
+            en_path = Path(LOCALES_DIR / "en" / filename)
+            if en_path.exists():
+                with open(en_path) as f:
+                    en_data = json.load(f)
+                file_total = sum(1 for _ in flatten_keys(en_data))
+            else:
+                file_total = 0
+            if file_path.exists():
+                with open(file_path) as f:
+                    data = json.load(f)
+                file_missing = missing_counts.get(filename, {}).get(lang, 0)
+                file_extra = len(summary.get(filename, LocaleSummary({}, {})).extra_keys.get(lang, []))
+
+                # Count TODOs
+                def count_todos(obj):
+                    if isinstance(obj, dict):
+                        return sum(count_todos(v) for v in obj.values())
+                    if isinstance(obj, list):
+                        return sum(count_todos(v) for v in obj)
+                    if isinstance(obj, str) and obj.strip().startswith("TODO: translate"):
+                        return 1
+                    return 0
+
+                file_todos = count_todos(data)
+                file_translated = file_total - file_missing
+                percent = 100 * file_translated / file_total if file_total else 100
+                style = (
+                    "bold green"
+                    if file_missing == 0 and file_extra == 0 and file_todos == 0
+                    else (
+                        "yellow" if file_missing < file_total or file_extra > 0 or file_todos > 0 else "red"
+                    )
+                )
+            else:
+                file_missing = file_total
+                file_extra = len(summary.get(filename, LocaleSummary({}, {})).extra_keys.get(lang, []))
+                file_todos = 0
+                file_translated = 0
+                percent = 0
+                style = "red"
+            table.add_row(
+                f"[bold reverse]{filename}[/bold reverse]",
+                str(file_missing),
+                str(file_extra),
+                str(file_todos),
+                str(file_translated),
+                str(file_total),
+                f"{percent:.1f}%",
+                style=style,
+            )
+            total_missing += file_missing
+            total_extra += file_extra
+            total_todos += file_todos
+            total_translated += file_translated
+            total_total += file_total
+        percent = 100 * total_translated / total_total if total_total else 100
+        table.add_row(
+            "All files",
+            str(total_missing),
+            str(total_extra),
+            str(total_todos),
+            str(total_translated),
+            str(total_total),
+            f"{percent:.1f}%",
+            style="red" if percent < 100 else "bold green",
+        )
+    for _lang, table in tables.items():
```

Comment:
Maybe in the loop you should have skipped "en" :-D as this is always 100%

Reply:
Feel free :)